Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,342 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers;
+
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Connection\AbstractConnection;
+use PhpAmqpLib\Connection\AMQPStreamConnection;
+use PhpAmqpLib\Exception\AMQPIOException;
+use PhpAmqpLib\Message\AMQPMessage;
+use PhpAmqpLib\Wire\AMQPTable;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessage;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\Context;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
+use Smartbox\Integration\FrameworkBundle\Service;
+
+/**
+ * Class PhpAmqpLibDriver.
+ */
+class PhpAmqpLibDriver extends Service implements AsyncQueueDriverInterface
+{
+    use UsesSerializer;
+
+    /**
+     * This field specifies the prefetch window size in octets.
+     * The server will send a message in advance if it is equal to or smaller in size than the available prefetch size
+     * (and also falls into other prefetch limits).
+     * May be set to zero, meaning "no specific limit", although other prefetch limits may still apply.
+     * The prefetch-size is ignored if the no-ack option is set.
+     */
+    const PREFETCH_SIZE = null;
```

Comment:
What does it mean if prefetch size is null? I think we should move this when defining the driver

Reply:
This param is used on the function basic_qos to define the max size of the message. That why  I created a constant to try to define a standard behavior for this function.

Reply:
Maybe I ask in a wrong way, if you setup PREFETCH_SIZE = null in the basic_qos what is going to happen?

Reply:
Here we talked to add as comment the reason why is null that was because of an ancient bug, but I don't see that comment

Reply:
@andreskrey or @brunohsouza can you add the comment explaining why this constant needs to be null? I remember you were mentioning there is a bug

Reply:
@brunohsouza this one is yours.

Reply:
The bug is happening when the prefetchSize value is different than 0. This argument is not implemented in Rabbitmq according this specification: https://www.rabbitmq.com/specification.html#method-status-basic.qos. 
I set this in the comment for the constant.

Reply:
:+1: 

Not sure why we want to keep the constant though, if its not configurable, nor we care what we do. I guess it came from the driver examples in the library?

Reply:
No. It doesn't came from the examples in the library. But, in an idea to use this in the future when there will be the possibility to configure this. 

If you want to remove it, that's ok. But, I think we should keep t as it is now.

Reply:
I think we can leave as it is, at least we have a comment explaining what is and why is null, that I think is better that just add null in `basic_qos` function